schemaregistry/schemaregistry.md

```diff
@@ -0,0 +1,318 @@
+
+## 2. Schema Registry Elements
+
+This section further describes the elements enumerated in the introduction.
+
+### 2.1. Schema Group
+
+A schema group is a named collection of schemas. Since a schema registry is
+often a resource with a scope greater than a single application and might even
+span multiple organizations, it is very useful to put a grouping construct
+around sets of schemas that are related either by ownership or by a shared
+subject matter context.
+
+Implementations of this specification MAY associate access control rules with
+schema groups. For instance, a user or a group of users might be given write
+access only to a particular schema group that their organization owns. If trade
+secret protection is required for an application or parts of it and the schema
+structure would give some of those away, read access to a group of schemas might
+likewise be restricted. 
+
+Access control rules at the schema group level MUST NOT limit visibility of the
+schema groups themselves to users authorized to enumerate schema groups at the
+registry level or to create new schema groups in order to create unambiguous
+transparency about which `id` have already been taken. An implementation MAY
+withhold all details other than the `id` attribute from a user not authorized
+to access the schema group.
+
+This specification does not define management constructs for such access control
+rules.
+
+### 2.1.1. Schema group attributes
+
+The data model for a schema group consists of three attributes:
+
+#### id
+
+- Type: `String`
+- Description: Identifies the schema group.
+- Constraints:
+  - REQUIRED
+  - MUST be a non-empty string
+  - MUST conform with RFC3986/3.3 `segment-nz-nc` syntax. This allows for "dot
+    notation", e.g. `org.example.myapp.module` for logical organization of
+    schema groups, even though the namespace is flat.
+  - MUST be unique within the scope of the schema registry
+- Examples:
+  - mygroup
+  - my-group
+  - org.example.myapp.module
+  - events@example.com
+
+#### format
+
+- Type: `String`
+- Description: Defines the schema format managed by this schema group. The
+  formats supported by an implementation are specific to the implementation. If
+  the format is omitted for the schema group, the format MUST be specified at
+  the schema-level. This specification does not mandate support for particular
+  formats.
+- Constraints:
+  - OPTIONAL.
+  - MUST be a non-empty string, if present.
+  - MUST NOT be modified if at least one schema exists in the group.
+- Examples:
+  - "avro"
+  - "json-schema"
+  - "xsd11"
+
+#### description
+
+- Type: `String`
+- Description: Explains the purpose of the schema group.
+- Constraints:
+  - OPTIONAL
+- Examples:
+  - "This group holds schemas for the fabulous example app."
+
+#### createdtimeutc
+
+- Type: `Timestamp`
+- Description: Instant when the schema group was added to the registry.
+- Constraints:
+  - OPTIONAL
```

Comment:
Should probably add a MUST in here about the format/syntax

Reply:
Pointing to the core spec.